platform-sdk/swirlds-virtualmap/src/main/java/com/swirlds/virtualmap/internal/merkle/VirtualRootNode.java

```diff
@@ -1520,6 +1532,7 @@ public void warm(final K key) {
     private void add(final K key, final V value) {
         throwIfImmutable();
         assert !isHashed() : "Cannot modify already hashed node";
+        assert currentModifyingThreadRef.compareAndSet(null, Thread.currentThread());
```

Comment:
This method is only called from `put()`, which already has the assertion. No need to assert anything here.

Reply:
Removed